### PR TITLE
Completed initial TW review for topic "Run a pipeline"

### PR DESCRIPTION
**Pull request by sen-sam**

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 1]
```diff
@@ -4,77 +4,101 @@ description = "Execute a pipeline on the KFP backend"
 weight = 6
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:4]: Should this be `title = "Run a pipeline"` to be consistent with "Compile a pipeline"?

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 2]
```diff
@@ -4,77 +4,101 @@ description = "Execute a pipeline on the KFP backend"
...
 +++
 
-The KFP SDK offers three ways to run a pipeline.
+You can run a pipeline in three ways using the KFP SDK:
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:7]:
```suggestion
You can run a pipeline in three ways:
```

Option 1 does not use the KFP SDK

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 3]
```diff
@@ -4,77 +4,101 @@ description = "Execute a pipeline on the KFP backend"
...
-  class="mt-3 mb-3 border border-info rounded">
+    <img src="/docs/images/pipelines/submit-a-pipeline-on-dashboard.png" alt="Upload pipeline button" class="mt-3 mb-3 border border-info rounded">
 
-3. Upload the pipeline IR YAML to "Upload a file", populate the upload pipeline form, and click "Create".
+<!-- TODO: This section needs more work -->
 
-<img src="/docs/images/pipelines/upload-a-pipeline.png" 
-  alt="Upload pipeline screen"
-  class="mt-3 mb-3 border border-info rounded">
+1. On the **Upload Pipeline or Pipeline Version** page, click **Upload a file** to upload the compiled IR YAML definition of your pipeline.
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:31]: numbering is showing up incorrectly

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 4]
```diff
@@ -4,77 +4,101 @@ description = "Execute a pipeline on the KFP backend"
...
 
-For example, the following command submits the `path/to/pipeline.yaml` IR YAML to the KFP backend:
+For example, run the following command to submit the IR YAML file at `path/to/pipeline.yaml` to the KFP backend:
 
 ```shell
 kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml
 ```
 
-For more information about the `kfp run create` command, see the [KFP Command Line Interface reference documentation][kfp-run-create-reference-docs]. For more information on the KFP CLI generally see [Command Line Interface user docs][kfp-cli].
+For more information about the `kfp run create` command, see [create][kfp-run-create-reference-docs] in the [KFP SDK API reference][kfp-sdk-api-ref]. For more information on the available commands in the KFP CLI, see [Command Line Interface][kfp-cli] in the [KFP SDK API reference][kfp-sdk-api-ref].
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:98]: nit:
`see [create]` -> `see [run create]`

The `create` action can be used for multiple resources, so probably want to be specific here.

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 5]
```diff
@@ -4,77 +4,101 @@ description = "Execute a pipeline on the KFP backend"
...
 
 ```shell
 kfp run create [OPTIONS] [ARGS]...
 ```
 
-For example, the following command submits the `path/to/pipeline.yaml` IR YAML to the KFP backend:
+For example, run the following command to submit the IR YAML file at `path/to/pipeline.yaml` to the KFP backend:
 
 ```shell
 kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:95]: Perhaps we should add an example of how to use `ARGS` to be consistent with the KFP SDK client example above:

```suggestion
kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml param="a" other_param=2
```

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]
Thanks, @sen-sam!

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md (hunk 1 above)
Comment (sen-sam) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:4]: Assuming that the mention of "backend" is not essential, accepting the comment.

Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:4]: Apologies if this was unclear. This wasn't a comment about the subtitle. I like it as it was before.

This was just whether the "p" of "pipeline" in "Run a Pipeline" should be lowercase to be consistent with the other change you suggested here: https://github.com/kubeflow/website/pull/3393/files#diff-f18c20da363f8a67e27396a3afab90fc9f260eebb2571e9473526bba8c10763eL2

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 6]
```diff
@@ -1,80 +1,107 @@
...
-The KFP SDK offers three ways to run a pipeline.
+You can run a pipeline in three ways:
 
-## 1. Run from the KFP Dashboard
-The first and easiest way to run a pipeline is by submitting it via the KFP dashboard.
+* [Run a pipeline from the KFP Dashboard](#run_from_dashboard)
+
+* [Run a pipeline from the KFP SDK client](#run_from_sdk_client)
+
+* [Run a pipeline from the KFP SDK CLI](#run_from_sdk_cli)
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:13]: The links for the second two bullets are not working for me. Can you check if they are for you? If not, perhaps Markdown-native section references (just linking to the title without HTML) will help with this?

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 7]
```diff
@@ -1,80 +1,107 @@
...
+description = "Run a pipeline"
 weight = 6
 +++
 
-The KFP SDK offers three ways to run a pipeline.
+You can run a pipeline in three ways:
 
-## 1. Run from the KFP Dashboard
-The first and easiest way to run a pipeline is by submitting it via the KFP dashboard.
+* [Run a pipeline from the KFP Dashboard](#run_from_dashboard)
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:9]: nit: can maybe remove "Run a pipeline" prefix from each of these for brevity

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 8]
```diff
@@ -1,80 +1,107 @@
...
-client.create_run_from_pipeline_func('pipeline.yaml', arguments={'param': 'a', 'other_param': 2})
-```
+To submit an IR YAML for execution:
+
+1. Instantiate a `Client` by specifying the `host` URL of your KFP instance:
+    ```python
+    from kfp.client import Client
+    client = Client(host='<YOUR_HOST_URL>')
+    ```
+    For more information about the `Client` constructor and method parameters, see the  [kfp.client][kfp-sdk-api-ref-client] in the [KFP SDK API reference][kfp-sdk-api-ref].
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:65]: nit: use backticks for code style for code symbols: `kfp.client`

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 9]
```diff
@@ -1,80 +1,107 @@
...
+1. Instantiate a `Client` by specifying the `host` URL of your KFP instance:
+    ```python
+    from kfp.client import Client
+    client = Client(host='<YOUR_HOST_URL>')
+    ```
+
+1. Use the `.create_run_from_pipeline_func` convenience method. This wraps the compilation and run submission into one method.
+
+    ```python
+    client.create_run_from_pipeline_func('pipeline.yaml', arguments={'param': 'a', 'other_param': 2})
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:84]: not your work: Just noticed `.create_run_from_pipeline_func('pipeline.yaml'...)` should be `.create_run_from_pipeline_func(my_pipeline)`. This was my mistake!

While you're making other changes, could you update this please?

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 10]
```diff
@@ -1,80 +1,107 @@
...
-For example, the following command submits the `path/to/pipeline.yaml` IR YAML to the KFP backend:
+For example, run the following command to submit the IR YAML file at `path/to/pipeline.yaml` to the KFP backend:
 
 ```shell
-kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml
+kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml param="a" other_param=2
 ```
 
-For more information about the `kfp run create` command, see the [KFP Command Line Interface reference documentation][kfp-run-create-reference-docs]. For more information on the KFP CLI generally see [Command Line Interface user docs][kfp-cli].
+For more information about the `kfp run create` command, see [run create][kfp-run-create-reference-docs] in the [KFP SDK API reference][kfp-sdk-api-ref]. For more information on the available commands in the KFP CLI, see [Command Line Interface][kfp-cli] in the [KFP SDK API reference][kfp-sdk-api-ref].
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:101]: nit: [`run create`] in backticks

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md [hunk 11]
```diff
@@ -1,80 +1,107 @@
...
 ```shell
 kfp run create [OPTIONS] [ARGS]...
 ```
 
-For example, the following command submits the `path/to/pipeline.yaml` IR YAML to the KFP backend:
+For example, run the following command to submit the IR YAML file at `path/to/pipeline.yaml` to the KFP backend:
 
 ```shell
-kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml
+kfp run create --experiment-name my-experiment --package-file path/to/pipeline.yaml param="a" other_param=2
```
Comment (connor-mccarthy) [inline content/en/docs/components/pipelines/v2/run-a-pipeline.md:98]: Should we update this example and placeholder style to be consistent with the presentation style from https://github.com/kubeflow/website/pull/3392/files#diff-20420e55f370813ad976788743b9900fa27c72eea0ee22d9a24a72940daf9a4eR114?

**Review by connor-mccarthy — CHANGES_REQUESTED** [review]

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md (hunk 7 above)
Comment (sen-sam) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:9]: My suggestion would be to remove "a pipeline" and retain "Run".

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md (hunk 9 above)
Comment (sen-sam) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:84]: Fix applied. Also removed the arguments.

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md (hunk 11 above)
Comment (connor-mccarthy) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:98]: Discussed offline -- can ignore this comment :)

File: content/en/docs/components/pipelines/v2/run-a-pipeline.md (hunk 1 above)
Comment (sen-sam) [reply content/en/docs/components/pipelines/v2/run-a-pipeline.md:4]: That makes sense. Fixed it.

**Comment by varodrig** [comment]
@hbelmiro  it seems we should close this PR since it the file doesn't longer exist.
cc @sen-sam

**Pull request closed without merging**